code/espurna/ota_httpupdate.ino

```diff
@@ -30,6 +31,41 @@ Copyright (C) 2019 by Maxim Prokhorov <prokhorov dot max at outlook dot com>
 
 #endif // SECURE_CLIENT != SECURE_CLIENT_NONE
 
+template <typename T>
+void _otaFollowRedirects(const std::true_type&, T& instance) {
+    instance.followRedirects(true);
+}
+
+template <typename T>
+void _otaFollowRedirects(const std::false_type&, T& instance) {
+}
+
+template <typename T>
+t_httpUpdate_return _otaClientUpdate(const std::true_type&, T& instance, WiFiClient* client, const String& url) {
+    if (client == nullptr) {
+        client = std::make_unique<WiFiClient>().get();
+    }
```

Comment:
btw shouln't this be back where it was initially? check for `has_WiFIClient_argument_t` there and supply it when true.
i don't remember the specifics (search is not helping...), but there were also a couple of issues with httpclient & wificlient destructor ordering. but, this may not be relevant if httpclient can handle wificlient missing and update() does not keep lingering connection / expects more actions.

Reply:
I moved it inside the function because that means one place less to check the update() signature: https://github.com/xoseperez/espurna/pull/1974/files#diff-5727c97832f80d3ac6aedd50dbe75fd6L81-L90 - but yes, not sure if this is the best approach.

Reply:
Note the second comment, WiFiClient is destroyed right after get().

Reply:
Hmm, that was not the case previously when it was called in the ``_otaClientFromHttp()`` function? I tested OTA with these changes and I didn't run into any issues so far.

Reply:
Not the case, unique_ptr lived while parent function was active.
idk how that works exactly... effectively, wificlient is no longer allocated and anything can claim it through new / malloc

Reply:
Ah, I guess it "accidentally" works because the pointer is used immediately after the .get() call. I think the best is to always create the WifiClient pointer? I'll commit it now

Reply:
Right. It could still check for arg availability though. Only cost is +2 lines, everything is known at compile time.

Reply:
So split ``_otaClientFromHttp()`` method and call the appropriate one from ``_otaClientFrom()`` with the type check? Or you mean a different way?

Reply:
I meant right in the fromhttp, optionally creating wificlient. unique_ptr structure could be initialized with nullptr, `.get()` will return that null pointer.

Reply:
Sorry bu I'm not sure if I follow - feel free to commit what you have in mind :)

Reply:
https://godbolt.org/z/yu9fxq <-
Maybe I misunderstood the question. `make_unique<T>() == unique_ptr<T>(new T())`, so we just hide make_unique under the check